scripts/irods/test/test_izonereport.py

```diff
@@ -86,14 +92,108 @@ def test_izonereport_and_validate(self):
+        
+        self.assertIn('coordinating_resources', zone_info.keys())
+        self.assertEqual(len(zone_info['coordinating_resources']), 3)
+        
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python', "only run for native rule language")
+    def test_zone_report_calculates_checksums_for_all_plugins__issue_6856(self):
+        
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        
+        for server in zone_info['servers']:
+            for plugin in server['plugins']:
+                
+                # Skip database plugin which only exists on catalog server when running on consumer server
+                if plugin['type'] == 'database' and test.settings.TOPOLOGY_FROM_RESOURCE_SERVER:
+                    continue
+                
+                plugin_name = f'lib{plugin["name"]}.so'
+                
+                # Find folder containing plugin file
+                plugin_folder = ""
+                for root, _dirs, files in os.walk(paths.plugins_directory()):
+                    if plugin_name in files:
+                        plugin_folder = root
+                        break
+                    
+                plugin_path = os.path.join(plugin_folder, plugin_name)
+                plugin_checksum = lib.file_digest(plugin_path, 'sha256', encoding='base64')
+                
+                self.assertEqual(plugin['checksum_sha256'][5:], plugin_checksum)
+        
+    def test_resource_json_has_comments_and_info_fields__issue_3739(self):
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        
+        for resource in zone_info['coordinating_resources']:
+            self.assertIn('comments', resource.keys())
+            self.assertIn('info', resource.keys())
+        
+        for server in zone_info["servers"]:
+            for resource in server['resources']:
+                self.assertIn('comments', resource.keys())
+                self.assertIn('info', resource.keys())
+
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing or python rule engine")
+    def test_no_server_duplication_core__issue_3682(self):
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        
+        self.assertEqual(len(zone_info['servers']), 1)
+    
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or not test.settings.RUN_IN_TOPOLOGY, "Skip for Core Testing or python rule engine")
+    def test_no_server_duplication_topology__issue_3682(self):
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        
+        self.assertEqual(len(zone_info['servers']), 4)
```

Comment:
You could detect the type of test being run here and use a different value instead of having two tests.

For instance:
```python
self.assertEqual(len(zone_info['servers']), 4 if test.settings.RUN_IN_TOPOLOGY else 1)
```

Same for `test_servers_are_flattened_topology__issue_6857` and `test_servers_are_flattened_core__issue_6857`

Reply:
Good point. I'll update the tests.